lib/rot.rb

```diff
@@ -0,0 +1,38 @@
+class Rot
+  ALPHABET = (("a".."z").to_a + ("A".."Z").to_a).freeze
+  MAX_ROT_KEY = 26
+
+  def initialize(rot_key)
+    @rot_key = rot_key
+    if @rot_key > MAX_ROT_KEY
+      raise ArgumentError, "Given rot key must be less than 26."
+    end
+  end
+
+  private
+  
+  def encryptable_letter?(letter)
+    ALPHABET.include?(letter)
+  end
+
+  def encrypt_letter(letter)
+    if encryptable_letter?(letter) 
+      @rot_key.times do 
+      letter = letter.next
+      end
+    end
+    letter
+  end
+
+  public
+
+  def encrypt(message)
+    message = message.split("")
+    result = ""
+    for letter in message
```

Comment:
@djordje997 generally in Ruby you should avoid for/while loops, instead use different kinds of iterators that are available. Most of the time objects will have `.each` iterator. In this case, `String` has `.each_char` (see https://ruby-doc.org/core-3.1.0/String.html#method-i-each_char).

You can also eliminate the split call by using an iterator.

```suggestion
    result = ""
    message.each_char do |letter|
```